### PR TITLE
Create blacklist_MOD.js

### PR DESCRIPTION
**Pull request by RigidStudios**
**Please describe the changes this PR makes and why it should be merged:**
Per-user global command blacklist, goes hand in hand with https://github.com/dbm-network/events/pull/21
Co-authored-by: noahalma

**Status**

- [X] Code changes have been tested against the Discord API and the discord.js wrapper, or there are no code changes

**Semantic versioning classification:**

- [x] This PR changes DBM's interface (methods or parameters added to default methods)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: actions/blacklist_MOD.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  init: function() {
+    const { document, glob } = this;
+  },
+  action: function(cache) {
+    const data = cache.actions[cache.index];
+    const varName = this.evalMessage(data.varName, cache);
+    const storage = parseInt(data.storage);
+    const type = parseInt(data.type);
+    const user = this.getVariable(storage, varName, cache)
+    let xxxTentaction;
```
Comment (Almeeida) [inline actions/blacklist_MOD.js:42]: really?

File: actions/blacklist_MOD.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    const info = ["", "Blacklist User", "Un-Blacklist User"];
+    const vars = ["", "Temp Variable", "Server Variable", "Global Variable"];
+    return `${vars[parseInt(data.storage)]}: ${data.varName} | ${info[parseInt(data.type)]}`;
+  },
+  fields: ["storage", "varName", "type"],
+  html: function(isEvent, data) {
+    return `
+  <div>
+    <div>
+    <p><u>Made by Rigid and Noah</u></p>
```
Comment (Almeeida) [inline actions/blacklist_MOD.js:14]: ❌

File: actions/blacklist_MOD.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        xxxTentaction = fs.readFileSync('./data/blacklist.txt', 'utf-8').toString();
+        users = xxxTentaction.split("\n");
+        if (users.includes(user.id)) {
+          console.log(users);
+          console.log(users.filter(x => x !== user.id));
+          fs.writeFileSync('./data/blacklist.txt', users.filter(x => x !== user.id).join("\n"));
+        }
+        break;
+    default:
+        console.log("wha- hwatt???");
```
Comment (Almeeida) [inline actions/blacklist_MOD.js:67]: a descriptive error message would be more useful

File: actions/blacklist_MOD.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        break;
+    }
+  },
+	  mod: function(DBM) {
+    const fs = require('fs');
+    if (!fs.existsSync('./data/blacklist.txt')) {
+      fs.writeFileSync('./data/blacklist.txt', "", function (err) {
+        if (err) {
+          console.log("Something went wrong...");
```
Comment (Almeeida) [inline actions/blacklist_MOD.js:76]: log the error?

File: actions/blacklist_MOD.js (hunk 3 above)
Comment (RigidStudios) [reply actions/blacklist_MOD.js:65]: Descriptive error message as in: that option literally isn't possible.

File: actions/blacklist_MOD.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    const info = ["", "Blacklist User", "Un-Blacklist User"];
+    const vars = ["", "Temp Variable", "Server Variable", "Global Variable"];
+    return `${vars[parseInt(data.storage)]}: ${data.varName} | ${info[parseInt(data.type)]}`;
+  },
+  fields: ["storage", "varName", "type"],
+  html: function(isEvent, data) {
+    return `
+  <div>
+    <div>
+    <p><u>Made by Rigid and Noah</u></p>
```
Comment (Almeeida) [inline actions/blacklist_MOD.js:14]:
```suggestion
```

File: actions/blacklist_MOD.js [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+      </select><br>
+      <div style="float: left; width: 47%">
+        Variable Type:<br><select id="storage" class="round">
+          ${data.variables[1]}
+        </select>
+      </div>
+      <div style="float: right; width: 47%">
+        Variable Name:<br>
+        <input class="round" id="varName" />
+</div>
```
Comment (Almeeida) [inline actions/blacklist_MOD.js:28]:
```suggestion
      </div>
```

File: actions/blacklist_MOD.js [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    </div>
+  </div>
+    `
+  },
+  init: function() {
+    const { document, glob } = this;
+  },
```
Comment (Almeeida) [inline actions/blacklist_MOD.js:35]:
```suggestion
  init: function() {},
```

File: actions/blacklist_MOD.js (hunk 5 above)
Comment (RigidStudios) [reply actions/blacklist_MOD.js:14]: DUDE IT WAS MADE BY BOTH OF US :KEKW:

**Pull request merged**